src/panels/lovelace/cards/hui-entities-card.ts

```diff
@@ -0,0 +1,191 @@
+import { HassLocalizeLitMixin } from "../../../mixins/lit-localize-mixin";
+import { LovelaceCard, LovelaceConfig } from "../types.js";
+import { HomeAssistant } from "../../../types.js";
+import createRowElement from "../common/create-row-element.js";
+import computeDomain from "../../../common/entity/compute_domain.js";
+import processConfigEntities from "../common/process-config-entities";
+
+interface EntityConfig {
+  name?: string;
+  icon?: string;
+  entity: string;
+  type?: string;
+  secondary_info: "entity-id" | "last-changed";
+  action_name?: string;
+  service?: string;
+  service_data?: object;
+  url?: string;
+}
+
+interface Config extends LovelaceConfig {
+  show_header_toggle?: boolean;
+  title?: string;
+  entities: EntityConfig[];
+}
+
+class HuiEntitiesCard extends HassLocalizeLitMixin(LitElement)
+  implements LovelaceCard {
+  protected hass?: HomeAssistant;
+  protected config?: Config;
+  protected configEntities?: EntityConfig[];
+
+  static get properties(): PropertyDeclarations {
+    return {
+      hass: {},
+      config: {},
+    };
+  }
+
+  public getCardSize() {
+    // +1 for the header
+    if (!this.config) {
+      return 0;
+    }
+    return this.config.title ? 1 : 0 + this.config.entities.length;
+  }
+
+  public setConfig(config: Config) {
+    const entities = processConfigEntities(config.entities);
+    for (const entity of entities) {
+      if (
+        entity.type === "call-service" &&
+        (!entity.service ||
+          !entity.name ||
+          !entity.icon ||
+          !entity.service_data ||
+          !entity.action_name)
+      ) {
+        throw new Error("Missing required property when type is call-service");
+      } else if (
+        entity.type === "weblink" &&
+        (!entity.name || !entity.icon || !entity.url)
+      ) {
+        throw new Error("Missing required property when type is weblink");
+      }
+    }
+
+    this.config = config;
+    this.configEntities = entities;
+
+    if (this.hass) {
+      this.requestUpdate();
+    }
+  }
+
+  protected render() {
+    if (!this.config || !this.hass) {
+      return html``;
+    }
+    const { show_header_toggle, title } = this.config;
+
+    const entities = this.configEntities!.map((conf) => conf.entity);
+
+    return html`
+      ${this.renderStyle()}
+      <ha-card>
+        ${
+          !title && !show_header_toggle
+            ? html``
+            : html`
+            <div class='header'>
+              ${!title ? html`` : html`<div class="name">${title}</div>`}
+              ${
+                show_header_toggle === false
+                  ? html``
+                  : html`
+                  <hui-entities-toggle 
+                    .hass="${this.hass}" 
+                    .entities="${entities}"
+                  >
+                  </hui-entities-toggle>`
+              }
+            </div>`
+        }
+        <div id="states">
+          ${repeat<EntityConfig>(
```

Comment:
I've been reading up on the repeat directive and also followed some discussion in the Polymer slack. It seems that we should move away from using repeat unless the order can change . Since we no longer filter out unavailable entities, our order won't ever change anymore -> a normal map over configEntities should be enough.